File: DOTNET BACKEND/TransitHive/Dtos/PaymentDto.cs
using TransitHive.Models;

namespace TransitHive.Dtos
{
    public class PaymentDto
    {
        public int PaymentId { get; set; }
        public int BookingId { get; set; }
        public int UserId { get; set; }
        public int VendorId { get; set; }
        public string PaymentMethod { get; set; }
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public PaymentStatus Status { get; set; } = PaymentStatus.PENDING;
    }

   
}

File: SpringBootBackend/src/main/java/com/th/dto/PaymentDTO.java
package com.th.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PaymentDTO extends BaseDTO {
    private Long bookingId;
    private Long userId;
    private Long vendorId;
    private String paymentMethod;
    private BigDecimal amount;
    private LocalDateTime paymentDate;
    private PaymentStatus status;
    private String transactionId;
    public enum PaymentStatus {
        SUCCESS, FAILED, PENDING
    }
}
